packages/entity-server/src/routes/hierarchy/types.ts

```diff
@@ -29,31 +48,36 @@ export type ExtendedFieldFunctions = Readonly<
 
 type ExcludeCommonFields<T, U> = Omit<T, Extract<keyof T, keyof U>>;
 
+export type FlattableEntityFields = Pick<EntityFields, SimpleFieldKeys<EntityFields>>;
```

Comment:
Perhaps we could redefine `SimpleFieldKeys`as `SimpleFields` (or `PickSimple`), by removing `[keyof T]` in that type, and then define this as 

```suggestion
export type FlattableEntityFields = SimpleFields<EntityFields>;
```